src/styles/getTooltipStyles.js

```diff
@@ -0,0 +1,177 @@
+ * @param {Number} windowWidth - The width of the window.
+ * @param {Number} xOffset - The distance between the left edge of the window
+ *                           and the left edge of the wrapped component.
+ * @param {Number} yOffset - The distance between the top edge of the window
+ *                           and the top edge of the wrapped component.
+ * @param {Number} componentWidth - The width of the wrapped component.
+ * @param {Number} componentHeight - The height of the wrapped component.
+ * @param {Number} tooltipWidth - The width of the tooltip itself.
+ * @param {Number} tooltipHeight - The height of the tooltip itself.
+ * @returns {Object}
+ */
+export default function getTooltipStyles(
+    currentSize,
+    windowWidth,
+    xOffset,
+    yOffset,
+    componentWidth,
+    componentHeight,
+    tooltipWidth,
+    tooltipHeight,
+) {
+    // Determine if the tooltip should display below the wrapped component.
+    // If a tooltip will try to render within GUTTER_WIDTH logical pixels of the top of the screen,
+    // we'll display it beneath its wrapped component rather than above it as usual.
+    const shouldShowBelow = (yOffset - tooltipHeight) < GUTTER_WIDTH;
+
+    // Determine if we need to shift the tooltip horizontally to prevent it
+    // from displaying too near to the edge of the screen.
+    const horizontalShift = computeHorizontalShift(windowWidth, xOffset, componentWidth, tooltipWidth);
+
+    const tooltipVerticalPadding = spacing.pv1;
+    const tooltipFontSize = variables.fontSizeSmall;
+
+    return {
+        animationStyle: {
+            transform: [{
+                scale: currentSize,
+            }],
+        },
+        tooltipWrapperStyle: {
+            position: 'absolute',
+            backgroundColor: themeColors.heading,
+            borderRadius: variables.componentBorderRadiusSmall,
+            ...tooltipVerticalPadding,
+            ...spacing.ph2,
+
+            // Shift the tooltip down (+) by...
+            //
+            //   component height
+            // + (pointer height - 1)
+            //
+            // OR
+            //
+            // Shift the tooltip up (-) by...
+            //   font height
+            // + top padding
+            // + bottom padding
+            // + upward shift of the pointer (pointer height)
+            // + (pointer height - 1)
+            //
+            top: shouldShowBelow
+                ? componentHeight + (POINTER_HEIGHT - 1)
+                : -1 * (
+                    tooltipFontSize
+                    + tooltipVerticalPadding.paddingTop
+                    + tooltipVerticalPadding.paddingBottom
+                    + ((2 * POINTER_HEIGHT) - 1)
+                ),
+
+            // Shift the tooltip to the left by...
+            //   half the component's width
+            // + half the tooltip's width
+            // + any horizontal shift
+            left: (tooltipWidth / -2) + (componentWidth / 2) + horizontalShift,
```

Comment:
Might be clearer to just make the first value negative rather than divide by a negative number. 

It's slightly unclear why the first value is negative? Maybe we can add a sentence to explain where the tooltip will end up positioned based on this instead of the summary of the code in the above comment?